_server/api/rental.py

```diff
@@ -132,35 +99,127 @@ def get_average_rental():
     ).filter(
         RentalRange.universityId == request.args['universityId']
     ).filter(
-        RentalRange.rentToUniversityDistance >= min_distance_km
+        RentalRange.rentToUniversityDistance >= min_distance
     ).filter(
-        RentalRange.rentToUniversityDistance <= max_distance_km
-    )
+        RentalRange.rentToUniversityDistance <= max_distance
+    ) 
 
-    if property_types:
+    # Filter baed on propertyType, bathCount, bedCount. If none, automatically unfiltered.
+    if request.args.get('propertyType') is not None:
         queried_rentals = queried_rentals.filter(
-            Rental.propertyType.in_(property_types))
-    if bath_count:
+            Rental.propertyType.in_(PROPERTY_ALIAS_MAP[request.args.get('propertyType')])
+        )        
+    if request.args.get('bathCount') is not None:
         queried_rentals = queried_rentals.filter(
-            Rental.bathroomCount == bath_count)
-    if bed_count:
+            Rental.bathroomCount == request.args.get('bathCount')
+        )
+    if request.args.get('bedCount') is not None:
         queried_rentals = queried_rentals.filter(
-            Rental.bedroomCount == bed_count)
+            Rental.bedroomCount == request.args.get('bedCount')
+        )
 
     queried_rentals = queried_rentals.all()
 
-    # Calculating average average
+    # Calculating average.
     rentals_count = len(queried_rentals)
-    rentals_average = calculate_average_rent_per_room(queried_rentals)
 
-    return {
-        'rentalsCount': rentals_count,
-        'average': rentals_average,
-        'distance': max_distance_km
-    }
+    # Check if valid rental average.
+    valid_avg = [
+        rentals_count > 0,
+    ]
+
+    if all(valid_avg):
+        rentals_average = calculate_average_rent_per_room(queried_rentals)
+        # Successfully calculated filtered average.
+        return {
+            'rentalsCount': int(rentals_count),
+            'summary': float(rentals_average),
+            'distance': int(max_distance),
+            'metric': 'average'
+        }
+
+    # Check if prediction is available.
+    prd_valid = [
+        request.args.get('propertyType') is not None,
+        # request.args.get('bedCount') is not None,
+        # request.args.get('bathCount') is not None
+    ]
+    
+    """ Prediction starts here """    
+    if (not all(valid_avg)) and all(prd_valid):
```

Comment:
not all(valid_avg)) doesn't need to be checked again, as we're already returning something from previous condition.

Reply:
Right